sale_order_line_variant_description/models/sale_order_line.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2015-17 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.onchange("product_id")
+    def _onchange_product_id_warning(self):
+        res = super()._onchange_product_id_warning()
+        if self.product_id:
+            product = self.product_id.with_context(lang=self.order_id.partner_id.lang)
+            if product.variant_description_sale:
+                self.name = product.variant_description_sale
+        return res
```

Comment:
At least on 18.0, As we are changing the name of the line, we could use the logic of `_compute_name`, specifically, the method `_get_sale_order_line_multiline_description_sale`. Something like

```suggestion
    def _get_sale_order_line_multiline_description_sale(self):
        return self.product_id.variant_description_sale or super()._get_sale_order_line_multiline_description_sale()
```

What do you think @moylop260 and @luisg123v?

Reply:
https://github.com/odoo/odoo/blob/84ebf7fe5c0c7f9e3237430e5adc5b9601092611/addons/sale/models/sale_order_line.py#L362